Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -15,55 +15,54 @@
 
 package com.amazon.randomcutforest;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
 import com.amazon.randomcutforest.sampler.Weighted;
-import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.state.sampler.CompactSamplerState;
+import com.amazon.randomcutforest.state.store.PointStoreDoubleState;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
+public class RandomCutForestState {
 
-    public long entreesSeen;
+    private long entreesSeen;
 
-    public double lambda;
+    private double lambda;
 
-    public int numberOfTrees;
+    private int numberOfTrees;
 
-    public int sampleSize;
+    private int sampleSize;
 
-    public int dimensions;
+    private int dimensions;
 
-    public int outputAfter;
+    private int outputAfter;
 
-    public boolean storeSequenceIndexesEnabled;
+    private boolean storeSequenceIndexesEnabled;
 
-    public boolean compactEnabled;
+    private boolean compactEnabled;
 
-    public boolean centerOfMassEnabled;
+    private boolean centerOfMassEnabled;
 
-    public boolean parallelExecutionEnabled;
+    private boolean parallelExecutionEnabled;
```

Comment:
raise this for final decision

Reply:
We discussed this - I think we can separate data and execution config. But for now going with a catchall "state" may be easier? The use case is that it saves the  effort of a person/program to remember/decide what are the execution configs, and simply reuse them in deserialization. If the information is saved, we can choose to not use it or use to separately. If it is not saved, then we have no such choice.